onnxruntime/core/providers/cuda/cuda_provider_factory.cc

```diff
@@ -40,14 +43,16 @@ std::unique_ptr<IExecutionProvider> CUDAProviderFactory::CreateProvider() {
   info.cuda_mem_limit = cuda_mem_limit_;
   info.arena_extend_strategy = arena_extend_strategy_;
   info.cudnn_conv_algo = cudnn_conv_algo_search_;
+  info.do_copy_in_default_stream = do_copy_in_default_stream_;
   return onnxruntime::make_unique<CUDAExecutionProvider>(info);
 }
 
 std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_CUDA(OrtDevice::DeviceId device_id,
                                                                                OrtCudnnConvAlgoSearch cudnn_conv_algo_search = OrtCudnnConvAlgoSearch::EXHAUSTIVE,
                                                                                size_t cuda_mem_limit = std::numeric_limits<size_t>::max(),
-                                                                               ArenaExtendStrategy arena_extend_strategy = ArenaExtendStrategy::kNextPowerOfTwo) {
-  return std::make_shared<onnxruntime::CUDAProviderFactory>(device_id, cuda_mem_limit, arena_extend_strategy, cudnn_conv_algo_search);
+                                                                               ArenaExtendStrategy arena_extend_strategy = ArenaExtendStrategy::kNextPowerOfTwo,
+                                                                               bool do_copy_in_default_stream = true) {
```

Comment:
>true [](start = 112, length = 4)

default to false to align with existing behavior?

Reply:
I thought about it, but random failure like #4829 is not the right default option. I think it's better to be safe than sorry. This option should not be set to false unless some models that currently working are impacted.